startleft/api/controllers/cloudformation/cloudformation_controller.py

```diff
@@ -39,12 +44,36 @@ def cloudformation(cft_file: UploadFile = File(..., description="File that conta
                    ):
     # Add custom mapping provided by customer
     cf_mapping_files = paths.default_cf_mapping_files
-    if len(mapping_file.filename) != 0:
+    if mapping_file and len(mapping_file.filename) != 0:
         cf_mapping_files = [mapping_file.file]
 
     # Run client
     cli.inner_run(type=type, map=cf_mapping_files, otm='threatmodel.otm', name=name, id=id,
                   ir_map=paths.default_ir_map, recreate=1, irius_server=ApiConfig.get_iriusrisk_server(),
                   api_token=api_token, filename=[cft_file.file])
 
-    return RESPONSE_BODY
+    return RESPONSE_BODY_POST
+
+
+@router.put('/projects/{project_id}', status_code=RESPONSE_STATUS_CODE_PUT)
+def cloudformation(project_id: str,
+                   cft_file: UploadFile = File(..., description="File that contains the CloudFormation Template"),
+                   type: FileType = Form(..., description="Format of the CloudFormation Template"),
+                   name: str = Form(..., description="Name of the new project"),
+                   api_token: str = Header(None, description="IriusRisk API token"),
+                   mapping_file: UploadFile = File(None, description="File that contains the mapping between AWS "
+                                                                     "components and IriusRisk components. Providing "
+                                                                     "this file will completely override default values"
+                                                   )
+                   ):
+    # Add custom mapping provided by customer
+    cf_mapping_files = paths.default_cf_mapping_files
```

Comment:
Definitely it must be refactored the code. It has been tried to fix it with your suggestion but manual tests failed when using a custom mapping file inside the post request.
To change it would affect to overall behaviour so in my opinion it would be better to include it in a refactor ticket.

Reply:
The solution provided to this consists of a specific method that checks the file provided via endpoint and provides the default file otherwise.